subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -149,22 +149,15 @@ static int cmd_wifi_connect(const struct shell *shell, size_t argc,
 	struct net_if *iface = net_if_get_default();
 	static struct wifi_connect_req_params cnx_params;
 	char *endptr;
-	int idx = 3;
+	int idx = 2;
 
 	if (shell_help_requested(shell) || argc < 3) {
 		shell_help_print(shell, NULL, 0);
 		return -ENOEXEC;
 	}
 
-	cnx_params.ssid_length = strtol(argv[2], &endptr, 10);
-	if (*endptr != '\0' || cnx_params.ssid_length <= 2) {
-		shell_help_print(shell, NULL, 0);
-		return -ENOEXEC;
-	}
-
-	cnx_params.ssid = &argv[1][1];
-
-	argv[1][cnx_params.ssid_length + 1] = '\0';
+	cnx_params.ssid = argv[1];
+	cnx_params.ssid_length = strlen(argv[1]);
```

Comment:
The WiFi specifies SSID as being a byte array, not a string. Thus why you could have anything there that would make this strlen() not going right. That said, nor this shell module is able to get escaped input (tough it could be implemented) nor the below device drivers below are handling the ssid properly (ti, atmel... they all treat it as a string, which is wrong).

Reply:
Yes, this is what the function is doing here, converting the SSID shell string to a 'byte array', actually to the expected ssid/ssid_length attributes.
I assume the specification defines what is going in the air (SSID is a byte array in 802.11 probe response, beacons...), but how you manage it in software is implementation specific.